spec/unit/plugin_manager/utils/http_client_spec.rb

```diff
@@ -0,0 +1,101 @@
+# encoding: utf-8
+require "pluginmanager/utils/http_client"
+require "uri"
+
+describe LogStash::PluginManager::Utils::HttpClient do
+  subject  { described_class }
+
+  describe ".start" do
+    context "with ssl" do
+      let(:uri) { URI.parse("https://localhost:8888") }
+
+      it "requires ssl" do
+        expect(Net::HTTP).to receive(:start).with(uri.host, uri.port, hash_including(:use_ssl => true))
+        described_class.start(uri)
+      end
+    end
+
+    context "without ssl" do
+      let(:uri) { URI.parse("http://localhost:8888") }
+
+      it "doesn't requires ssl" do
+        expect(Net::HTTP).to receive(:start).with(uri.host, uri.port, hash_including(:use_ssl => false))
+        described_class.start(uri)
+      end
+    end
+  end
+
+  describe ".remove_file_exist?" do
+    let(:mock_http) { double("Net::HTTP") }
+
+    before do
+      expect(subject).to receive(:start).with(anything).and_yield(mock_http).at_least(:once)
```

Comment:
usually I prefer to use allow in `before` blocks and `expects` in assertions. any thoughts on that?